src/main/java/graphs/GraphSearch.java

```diff
@@ -0,0 +1,77 @@
+package graphs;
+
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Set;
+
+public class GraphSearch {
+
+    public List<Node> depthFirstSearch(Graph g) {
+
+        if(g == null || g.getNodes() == null || g.getEdges() == null) {
+            throw new IllegalArgumentException("empty or null graph");
+        }
+
+        UndirectedGraph graph = new UndirectedGraph(g);
+
+        // DFS uses a stack of nodes visited
+        // stack is push pop peek : peek to see
+        LinkedList<Node> locationStack = new LinkedList<Node>();
+        Node startingNode = ((Node)graph.getNodes().toArray()[0]);
```

Comment:
I have a problem with this because getNodes() returns a Set, not a List, so you're not guaranteed any ordering of Nodes. The depthFirstSearch method should take two additional arguments, Node start and Node end, that represent the start and end of the search criteria through the Graph.